contrib/nydusify/pkg/checker/rule/bootstrap.go

```diff
@@ -47,42 +38,5 @@ func (rule *BootstrapRule) Validate() error {
 		return errors.Wrap(err, "invalid nydus bootstrap format")
 	}
 
-	// For registry garbage collection, nydus puts the blobs to
-	// the layers in manifest, so here only need to check blob
-	// list consistency for registry backend.
-	if rule.BackendType != "registry" {
-		return nil
-	}
-
-	var bootstrap bootstrapDebug
-	bootstrapBytes, err := ioutil.ReadFile(rule.DebugOutputPath)
-	if err != nil {
-		return errors.Wrap(err, "read bootstrap debug json")
-	}
-	if err := json.Unmarshal(bootstrapBytes, &bootstrap); err != nil {
-		return errors.Wrap(err, "unmarshal bootstrap output JSON")
-	}
-
-	// Parse blob list from blob layers in Nydus manifest
-	var blobListInLayer []string
-	layers := rule.Parsed.NydusImage.Manifest.Layers
-	for i, layer := range layers {
-		if i != len(layers)-1 {
-			blobListInLayer = append(blobListInLayer, layer.Digest.Hex())
-		}
-	}
-
-	// Blob list recorded in manifest annotation should be equal with
-	// the blob list recorded in blob table of bootstrap
-	if !reflect.DeepEqual(bootstrap.Blobs, blobListInLayer) {
```

Comment:
We should still validate that all the blobs in bootstrap are in the image manifest, to make sure we don't reference any non-existing blobs in bootstrap.

Reply:
Good point! fixed.